src/platform/silabs/efr32/BLEChannelImpl.cpp

```diff
@@ -0,0 +1,447 @@
+    }
+    mAdvConnectableMode = connectMode;
+
+    ret = sl_bt_legacy_advertiser_start(mAdvHandle, mAdvConnectableMode);
+    VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BLEChannelImpl::OpenConnection(bd_addr address, uint8_t addrType)
+{
+    VerifyOrReturnError(addrType <= sl_bt_gap_random_nonresolvable_address, CHIP_ERROR_INVALID_ARGUMENT);
+    if (mConnectionState.allocated)
+    {
+        // We only support one connection at a time for now
+        ChipLogError(DeviceLayer, "Connection already allocated");
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    sl_status_t ret = sl_bt_connection_open(address, addrType, sl_bt_gap_1m_phy, &mConnectionState.connectionHandle);
+    VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
+
+    // TODO: Confirm this generates a connection event and the AddConnection gets called so that the connection state is updated
+    return CHIP_NO_ERROR;
+}
+CHIP_ERROR BLEChannelImpl::SetConnectionParams(const Optional<uint8_t> & connectionHandle, uint32_t intervalMin,
+                                               uint32_t intervalMax, uint16_t latency, uint16_t timeout)
+{
+    sl_status_t ret;
+    if (connectionHandle.HasValue())
+    {
+        ret = sl_bt_connection_set_parameters(connectionHandle.Value(), intervalMin, intervalMax, latency, timeout, 0, 0xFFFF);
+        VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
+    }
+    else
+    {
+        ret = sl_bt_connection_set_default_parameters(intervalMin, intervalMax, latency, timeout, 0, 0xFFFF);
+        VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BLEChannelImpl::SetAdvertisingParams(uint32_t intervalMin, uint32_t intervalMax, uint16_t duration,
+                                                const Optional<uint16_t> & maxEvents, const Optional<uint8_t> & channelMap)
+{
+    sl_status_t ret;
+
+    mAdvIntervalMin = intervalMin;
+    mAdvIntervalMax = intervalMax;
+    mAdvDuration    = duration;
+    mAdvMaxEvents   = maxEvents.ValueOr(0);
+
+    if (channelMap.HasValue())
+    {
+        ret = sl_bt_advertiser_set_channel_map(mAdvHandle, channelMap.Value());
+        VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
+    }
+
+    ret = sl_bt_advertiser_set_timing(mAdvHandle, mAdvIntervalMin, mAdvIntervalMax, mAdvDuration, mAdvMaxEvents);
+    VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BLEChannelImpl::CloseConnection()
+{
+    sl_status_t ret = sl_bt_connection_close(mConnectionState.connectionHandle);
+    VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
+    // Todo: Confirm this generates a disconnect event and the RemoveConnection gets called
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BLEChannelImpl::SetAdvHandle(uint8_t handle)
+{
+    sl_status_t ret;
+    if (mAdvHandle == 0xff)
```

Comment:
ca pourrait pas etre plus optimiser du genre 
```suggestion
    if (handle == mAdvHandle) 
    {
        return CHIP_NO_ERROR;
    }
    
    sl_bt_advertiser_delete_set(mAdvHandle); // No need to check since we are creating a new set anyway
    mAdvHandle = handle;
    return MapBLEError(sl_bt_advertiser_create_set(&mAdvHandle));
    
```

Pas sur si c'est mieux au niveau behaviour par contre